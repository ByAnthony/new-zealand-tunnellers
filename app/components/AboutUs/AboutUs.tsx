"use client";

import Image from "next/image";
import { AboutUsArticle } from "../../types/article";
import { Paragraph } from "../Article/Paragraph/Paragraph";
import { Title } from "../Title/Title";

import STYLES from "./AboutUs.module.scss";

type Props = {
  article: AboutUsArticle;
};

export function AboutUs({ article }: Props) {
  return (
    <div className={STYLES.container}>
      <div className={STYLES.header}>
        <Title title={article.title} />
      </div>
      <Paragraph section={article.section[0]} />
      <div className={STYLES["image-container"]}>
        <Image
          src={`/images/about-us/${article.image[0].file}`}
          alt={article.image[0].alt}
          width={800}
          height={575}
          className={STYLES.image}
        />
      </div>
      <Paragraph section={article.section[1]} />
    </div>
  );
}
