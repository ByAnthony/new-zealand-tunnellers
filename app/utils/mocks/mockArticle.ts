import { AboutUsArticle } from "../../types/article";

export const mockArticle: AboutUsArticle = {
  id: "about-us",
  title: "Our Story",
  section: [
    { title: "Title 1", text: "This is the first paragraph of our story." },
    { title: "Title 2", text: "This is the second paragraph of our story." },
  ],
  image: [
    {
      file: "our-story.jpg",
      title: null,
      photographer: null,
      reference: null,
      alt: "A picture of our team",
    },
  ],
};
