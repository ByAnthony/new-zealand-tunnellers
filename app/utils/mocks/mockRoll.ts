import { Tunneller } from '@/app/types/roll';
import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails: Tunneller = {
  id: mockId,
  name: johnDoe,
  birthDate: '1886',
  deathDate: '1952',
};

export const mockDetailsList: Tunneller[] = [
  mockDetails,
  {
    ...mockDetails,
    id: 1,
    name: alexanderDriscott,
  },
];

export const mockRoll: Record<string, Tunneller[]> = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
