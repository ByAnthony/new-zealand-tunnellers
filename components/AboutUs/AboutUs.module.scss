@use "app/variables.module.scss";

.container {
  @include variables.component-margin;

  margin-top: variables.$l-margin;
  margin-bottom: 40px;

  .header {
    margin-bottom: variables.$l-margin;

    @include variables.tablet {
      margin-top: variables.$xl-margin;
    }

    @include variables.desktop {
      margin-bottom: 125px;
    }
  }

  a {
    @include variables.text-link;

    font-size: 1.15rem;
    font-weight: 500;
    vertical-align: baseline;
    line-height: 100%;

    @include variables.desktop {
      font-size: 1.25rem;
      line-height: 2.25rem;
    }
  }

  @include variables.tablet {
    margin-top: variables.$xl-margin;
  }

  @include variables.desktop {
    margin-top: variables.$xxl-margin;
  }
}

.image-container {
  display: flex;
  margin: variables.$base-margin 0;
  max-width: variables.$inner-width;
  justify-content: center;
  align-items: center;

  @include variables.tablet {
    margin: variables.$l-margin 0;
  }

  @include variables.desktop {
    margin: 0 auto;
    margin-top: variables.$xl-margin;
    margin-bottom: variables.$xl-margin;
  }
}

.image {
  object-fit: cover;
  object-position: center;
  height: auto;
  width: 100%;
  border-radius: variables.$border-radius;

  @include variables.desktop {
    max-width: variables.$inner-width;
  }
}

.get-in-touch {
  margin-bottom: variables.$base-margin;

  @include variables.tablet {
    margin-bottom: variables.$l-margin;
  }

  @include variables.desktop {
    margin-bottom: variables.$l-margin;
  }
}
