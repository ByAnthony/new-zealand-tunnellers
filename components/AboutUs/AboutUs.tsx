"use client";

import Image from "next/image";
import { useEffect } from "react";

import { Paragraph } from "@/components/Article/Paragraph/Paragraph";
import { Title } from "@/components/Title/Title";
import { AboutUsArticle } from "@/types/article";

import STYLES from "./AboutUs.module.scss";

type Props = {
  article: AboutUsArticle;
};

export function AboutUs({ article }: Props) {
  useEffect(() => {
    window.localStorage.clear();
  }, []);

  return (
    <div className={STYLES.container}>
      <div className={STYLES.header}>
        <Title title={article.title} />
      </div>
      <Paragraph section={article.section[0]} />
      <div className={STYLES["image-container"]}>
        <Image
          src={`/images/about-us/${article.image[0].file}`}
          alt={article.image[0].alt}
          width={800}
          height={575}
          className={STYLES.image}
          placeholder="empty"
        />
      </div>
      <div className={STYLES["get-in-touch"]}>
        <Paragraph section={article.section[2]} />
        <div className={STYLES["contact-buttons"]}>
          <button
            type="button"
            className={STYLES.email}
            onClick={() => window.open("mailto:info@nztunnellers.com")}
            aria-label="Contact by email"
          >
            Email
          </button>
          <button
            type="button"
            className={STYLES.linkedin}
            onClick={() =>
              (window.location.href =
                "https://www.linkedin.com/in/anthony-byledbal/")
            }
            aria-label="Contact on LinkedIn"
          >
            LinkedIn
          </button>
        </div>
      </div>
      <Paragraph section={article.section[1]} />
    </div>
  );
}
