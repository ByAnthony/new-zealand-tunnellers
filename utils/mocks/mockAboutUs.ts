import { AboutUsArticle } from "@/types/article";

import {
  mockSectionOne,
  mockSectionTwo,
  mockTitle,
  mockTopImage,
} from "./mockArticle";

export const mockAboutUs: AboutUsArticle = {
  id: "my-awesome-article",
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
