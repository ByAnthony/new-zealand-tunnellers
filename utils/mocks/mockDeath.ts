import { Death } from "@/types/tunneller";

import { mockDeath } from "./mockTunneller";

export const mockAfterWarDeath: Death = {
  ...mockDeath,
  warInjuriesDeathAfterWar: true,
};
