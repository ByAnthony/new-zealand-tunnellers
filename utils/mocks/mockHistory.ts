import { HistoryChapter } from "@/types/homepage";

export const mockHistory: HistoryChapter[] = [
  {
    id: "where-all-started",
    chapter: 1,
    title: "Where All\\Started",
    image: "where-all-started.jpg",
  },
  {
    id: "the-journey-ahead",
    chapter: 2,
    title: "The Journey\\Ahead",
    image: "the-journey-ahead.jpg",
  },
  {
    id: "discovering-the-war",
    chapter: 3,
    title: "Discovering\\The War",
    image: "discovering-the-war.jpg",
  },
  {
    id: "below-the-surface",
    chapter: 4,
    title: "Below\\The Surface",
    image: "below-the-surface.jpg",
  },
  {
    id: "back-to-the-surface",
    chapter: 5,
    title: "Back\\To The Surface",
    image: "back-to-the-surface.jpg",
  },
  {
    id: "reconversion-of-activities",
    chapter: 6,
    title: "Reconversion\\Of Activities",
    image: "reconversion-of-activities.jpg",
  },
  {
    id: "returnin-home",
    chapter: 7,
    title: "Returning\\Home",
    image: "returning-home.jpg",
  },
];
