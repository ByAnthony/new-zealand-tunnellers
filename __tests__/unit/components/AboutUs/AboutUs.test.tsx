import { fireEvent, render, screen } from "@testing-library/react";

import { AboutUs } from "@/components/AboutUs/AboutUs";
import { mockAboutUs } from "__tests__/unit/utils/mocks/mockAboutUs";

let originalWindowLocation = window.location;

beforeEach(() => {
  Object.defineProperty(window, "location", {
    configurable: true,
    enumerable: true,
    value: new URL(window.location.href),
  });
});

afterEach(() => {
  Object.defineProperty(window, "location", {
    configurable: true,
    enumerable: true,
    value: originalWindowLocation,
  });
});

describe("AboutUs", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<AboutUs article={mockAboutUs} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the component correctly", () => {
    render(<AboutUs article={mockAboutUs} />);

    const titleLineOne = screen.getByText("My Awesome");
    expect(titleLineOne).toHaveClass("title-line-1");

    const titleLineTwo = screen.getByText("Article Title");
    expect(titleLineTwo).toHaveClass("title-line-2");

    expect(screen.getByText("Section Title 1")).toBeInTheDocument();
    expect(screen.getByText("Section text one")).toBeInTheDocument();

    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("alt", "Accessible alt text");

    expect(screen.getByText("Section Title 2")).toBeInTheDocument();
    expect(screen.getByText("Section text two")).toBeInTheDocument();
  });

  test("should open mailto link when email button is clicked", () => {
    window.open = jest.fn();

    const { getByLabelText } = render(<AboutUs article={mockAboutUs} />);

    const emailButton = getByLabelText("Contact by email");

    fireEvent.click(emailButton);

    expect(window.open).toHaveBeenCalledWith("mailto:info@nztunnellers.com");
  });

  test("should navigate to LinkedIn URL when LinkedIn button is clicked", () => {
    const { getByLabelText } = render(<AboutUs article={mockAboutUs} />);

    const linkedinButton = getByLabelText("Contact on LinkedIn");

    fireEvent.click(linkedinButton);

    expect(window.location.href).toBe(
      "https://www.linkedin.com/in/anthony-byledbal/",
    );
  });
});
