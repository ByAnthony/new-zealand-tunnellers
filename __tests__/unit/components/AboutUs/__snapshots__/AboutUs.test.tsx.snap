// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="main-title"
      >
        <h1>
          <span
            class="title-line-1"
          >
            My Awesome
          </span>
          <span
            class="title-line-2"
          >
            Article Title
          </span>
        </h1>
      </div>
    </div>
    <div
      class="paragraph"
    >
      <h2>
        Section Title 1
      </h2>
      <p>
        <span>
          Section text one
        </span>
      </p>
    </div>
    <div
      class="image-container"
    >
      <img
        alt="Accessible alt text"
        class="image"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="575"
        src="/_next/image?url=%2Fimages%2Fabout-us%2Fimg-123.png&w=1920&q=75"
        srcset="/_next/image?url=%2Fimages%2Fabout-us%2Fimg-123.png&w=828&q=75 1x, /_next/image?url=%2Fimages%2Fabout-us%2Fimg-123.png&w=1920&q=75 2x"
        style="color: transparent;"
        width="800"
      />
    </div>
    <div
      class="get-in-touch"
    >
      <div
        class="paragraph"
      >
        <h2>
          Section Title 3
        </h2>
        <p>
          <span>
            Section text three
          </span>
        </p>
      </div>
    </div>
    <div
      class="paragraph"
    >
      <h2>
        Section Title 2
      </h2>
      <p>
        <span>
          Section text two
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
